Sources/Utilities/Image Saver/ImageTools.swift

```diff
@@ -0,0 +1,301 @@
+      let heightRatio = targetSize.height / originalSize.height
+      let scaleFactor = min(widthRatio, heightRatio)
+      let newSize = CGSize(
+        width: originalSize.width * scaleFactor,
+        height: originalSize.height * scaleFactor
+      )
+      
+      let renderer = UIGraphicsImageRenderer(size: newSize)
+      let newImage = renderer.image { _ in
+        image.draw(in: CGRect(origin: .zero, size: newSize))
+      }
+      
+      return newImage
+    }.value
+  }
+  
+  /// Downsizes the given image by the specified percentage asynchronously.
+  /// - Parameters:
+  ///   - image: The original UIImage to be downsized.
+  ///   - percentage: The percentage by which the image should be downsized. Must be greater than 0 and less than or equal to 100.
+  /// - Returns: A UIImage that is the result of the downsizing operation.
+  ///
+  /// - Example:
+  /// ```swift
+  /// do {
+  ///   let image = UIImage(named: "exampleImage")!
+  ///   let percentage: CGFloat = 50.0
+  ///   let resizedImage = try await downsize(image, by: percentage)
+  ///   imageView.image = resizedImage
+  /// } catch {
+  ///   print("Failed to downsize image: \(error)")
+  /// }
+  /// ```
+  /// - Note: This function should be called from an asynchronous context using `await`.
+  /// - Throws: `ImageError.invalidPercentage` if the percentage is not within the valid range.
+  func downsize(_ image: UIImage, by percentage: CGFloat) async throws -> UIImage {
+    guard percentage > 0 && percentage <= 100 else {
+      throw ImageError.invalidPercentage
+    }
+    
+    return await Task.detached(priority: .high) {
+      let scaleFactor = percentage / 100.0
+      let newSize = CGSize(
+        width: image.size.width * scaleFactor,
+        height: image.size.height * scaleFactor
+      )
+      
+      let renderer = UIGraphicsImageRenderer(size: newSize)
+      let newImage = renderer.image { _ in
+        image.draw(in: CGRect(origin: .zero, size: newSize))
+      }
+      
+      return newImage
+    }.value
+  }
+  
+  /// Compresses the given image to the specified format and compression ratio.
+  ///
+  /// This function compresses the provided UIImage to the specified format (JPEG or PNG).
+  /// It returns the compressed image data.
+  /// If the compression operation fails, the function throws an error.
+  ///
+  /// - Parameters:
+  ///   - image: The UIImage to be compressed.
+  ///   - format: The desired image format (JPEG or PNG).
+  /// - Returns: The compressed image data as `Data`.
+  /// - Example:
+  /// ```swift
+  /// do {
+  ///   let image = UIImage(named: "exampleImage")!
+  ///   let compressedData = try compress(image, format: .jpeg(compressionRatio: 0.8))
+  /// } catch {
+  ///   print("Failed to compress image: \(error)")
+  /// }
+  /// ```
+  /// - Throws: `ImageError.compression` if the compression operation fails.
+  func compress(_ image: UIImage, format: ImageFormat) throws -> Data {
```

Comment:
Lets use swift syntax?:

```swift
func compress(image: UIImage, with format: ImageFormat)
```

Reply:
I agree this should be unified.

The call sites...
```swift
now
let data = compress(myImage, format: .png)

after
let data = compres(image: myImage, with: .png)
```

Your proposal changes first param to explicit, where the second is implicit/unclear.
I my case, the first is implicit, the second is explicit.
If you explicitly define one param, we should do the same for others.

With that in mind, you suggestion should be modified to
```swift
func compress(image: UIImage, withFormat: ImageFormat)
```

I would change it to
```swift
func compress(image: UIImage, format: ImageFormat)
// or
func compress(image: UIImage, withFormat format: ImageFormat)
```

The above solutions call site is as follows:
```swift
compress(image: myImage, format: .png)
compress(image: myImage, withFormat: .png)
```

Reply:
Perfect, completely agree, makes more sense 👏